Update Dashicons, assign new icons to embeds and default block

jasmussen opened the pull request:
## Description

This PR updates the Dashicons component to the latest version, and adds a number of new icons for Gutenberg:

- default block
- generic embed
- photo embed
- video embed
- audio embed
- post embed

Additionally, this PR assigns those icons to the default block, so if you don't provide an icon when registering a new block, you get a nice little similar-but-legally-distinct LEGO icon.

It also assigns the embed icons to all the embeds — generic for the genericon one, and the most topical one for the rest. This removes a lot of logo branding from the inserter, and simplifies the overall silhuette:

<img width="369" alt="screen shot 2017-10-16 at 10 26 22" src="https://user-images.githubusercontent.com/1204802/31602355-0bd599c8-b25d-11e7-86dd-29b6b85827da.png">

It frees us from having to add every other social logo to the icon sets in the future, but providing generic ones. 

For now, I'm going to say this fixes #2120, and we can then reopen this if we still find it doesn't.

## How Has This Been Tested?

Tested in Docker/Chrome.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

Commit:
Update dashicons component and change default icon

Commit:
Fix issue with dashicons component, assign embed icons.

gziolo commented:
Some JS tests are failing. I bet this is because you changed names of icons. Let me know if you need help with fixing those tests.

Soean commented on blocks/api/registration.js:
```diff
@@ -111,7 +111,7 @@ export function registerBlockType( name, settings ) {
 	}
 	if ( ! settings.icon ) {
-		settings.icon = 'star-filled';
+		settings.icon = 'block-default';
```
You also have to change the icon in the test file.

jasmussen commented:
Ah, thanks gziolo and Soean, will do some fixes.

Commit:
Update tests.

gziolo commented:
Travis is happy, you can ignore codecov check.

jasmussen commented:
Is that a 👍 👍 to merge? ;)

gziolo commented on blocks/library/embed/index.js:
```diff
@@ -342,175 +342,175 @@ registerBlockType(
 	getEmbedBlockSettings( {
 		title: 'Cloudup',
-		icon: 'cloud',
+		icon: 'embed-post',
```
Cloudup - shouldn't it be `embed-photo` based on the list from #2120?

jasmussen replied:
Since you can put video and markdown documents there too, I went with a more generic "post" icon.

gziolo commented on components/dashicon/index.js:
```diff
@@ -829,7 +856,6 @@ export default class Dashicon extends wp.element.Component {
 			<svg
 				aria-hidden
 				role="img"
-				focusable="false"
```
I don't know what implications it has, but if you are confident doing it then 👍

jasmussen replied:
Argh, comparing https://github.com/WordPress/gutenberg/blob/master/components/dashicon/index.js with the upstream https://github.com/WordPress/dashicons/blob/master/react/index.jsx reveals we have pushed commits to Gutenberg that weren't pushed back upstream. 

Pressing pause in this PR as I sync that stuff up.

gziolo commented on components/dashicon/index.js:
```diff
@@ -353,7 +359,7 @@ export default class Dashicon extends wp.element.Component {
 				path = 'M14.29 4.59l1.1 1.11c.41.4.61.94.61 1.47v2.12c0 .53-.2 1.07-.61 1.47l-6.63 6.63c-.4.41-.94.61-1.47.61s-1.07-.2-1.47-.61l-1.11-1.1-1.1-1.11c-.41-.4-.61-.94-.61-1.47v-2.12c0-.54.2-1.07.61-1.48l6.63-6.62c.4-.41.94-.61 1.47-.61s1.06.2 1.47.61zm-6.21 9.7l6.42-6.42c.39-.39.39-1.03 0-1.43L12.36 4.3c-.19-.19-.45-.29-.72-.29s-.52.1-.71.29l-6.42 6.42c-.39.4-.39 1.04 0 1.43l2.14 2.14c.38.38 1.04 ...
 				break;
 			case 'editor-rtl':
```
Double checking, was it intended to update this icon? `editor-ltr` was also introduced, so that might be the reason why.

jasmussen replied:
LTR was introduced, yes, and I would assume some paths in the RTL were also polished upstream.

gziolo reviewed (approved):
Code looks good, I can see all the icons updated in the UI:

![screen shot 2017-10-16 at 11 35 13](https://user-images.githubusercontent.com/699132/31605351-51d26c22-b266-11e7-982a-c2f4abd42dfa.png)


I left 2 more comments where I wasn't sure why those changes landed. If you think they should be part of this PR then you can 🚢

jasmussen commented:
Thanks for the approval, but I've sort of "unapproved" this myself — turns out there are some upstream changes we have to fix first.

Commit:
Fix upstream dashicons regression.

This will be fixed upstream when https://github.com/WordPress/dashicons/pull/241 is merged.

jasmussen commented:
Pushed a fix that includes the "focusable" tweak, which is also fixed upstream in https://github.com/WordPress/dashicons/pull/241. 

If the approval still stands, I'm going to merge when the checks pass. Thanks again.
